### PR TITLE
Correction filter when POST Request

### PR DESCRIPTION
**Pull request by Flounn**
- Correction filter (var from advancedSearch) with a POST request => if stringify, the variable is send like a string.
- Change event on the Search input and no key up when the sidePagination===server

### CODE REVIEW

File: src/bootstrap-table.js [hunk 1]
```diff
@@ -1178,28 +1179,37 @@
...
-                }
-
-                if ($.inArray(event.keyCode, [37, 38, 39, 40]) > -1) {
-                    return;
-                }
-
-                clearTimeout(timeoutId); // doesn't matter if it's 0
-                timeoutId = setTimeout(function () {
+            
+            if (that.options.sidePagination ==='server'){ // Event on the change : decrease the number of request to the server and more user friendly than a timer
```
Comment (djhvscf) [inline src/bootstrap-table.js]: Missing an space between last parenthesis and first bracket

File: src/bootstrap-table.js [hunk 2]
```diff
@@ -1178,28 +1179,37 @@
...
+                $search.off('change').on('change', function (event){
```
Comment (djhvscf) [inline src/bootstrap-table.js]: You are missing an space between last parenthesis and first bracket

File: src/bootstrap-table.js [hunk 3]
```diff
@@ -1968,7 +1978,7 @@
         }
 
         if (!($.isEmptyObject(this.filterColumnsPartial))) {
-            params.filter = JSON.stringify(this.filterColumnsPartial, null);
+            params.filter = this.options.method==="get"?JSON.stringify(this.filterColumnsPartial, null):this.filterColumnsPartial;//Don't stringify if it's a POST request : the data are Stringify after and the object is passing to a string...
```
Comment (djhvscf) [inline src/bootstrap-table.js]: Missing space `...method === "get" ? JSON`

File: src/bootstrap-table.js [hunk 4]
```diff
@@ -2474,6 +2484,9 @@
         if (len === this.options.data.length) {
             return;
         }
+        
+        if (this.options.sidePagination==='server')
```
Comment (djhvscf) [inline src/bootstrap-table.js]: Missing brackets

File: src/bootstrap-table.js [hunk 5]
```diff
@@ -916,6 +916,7 @@
         }
 
         if (this.options.sidePagination === 'server') {
+            this.options.totalRows=this.options.data.length;
```
Comment (djhvscf) [inline src/bootstrap-table.js]: Missing a space `...totalRow = this.options....`

**Comment by djhvscf** [comment]
Provide an example, please

**Comment by Flounn** [comment]
Example issue POST request :
- In the Advanced search I filter with firstname and lastname. See AdvancedSearch.JPG
- The params filter is stringify line 1959 and also line 1978 because I have a POST request. The params filter is an objet but with 2 stringify, it is send as a string in the body of the request. See request_before.txt
- With my correction, the first stringify is avoid and the params filter is send as an object. See request_after.txt
advancedsearch
[request_after.txt](https://github.com/wenzhixin/bootstrap-table/files/402902/request_after.txt)
[request_before.txt](https://github.com/wenzhixin/bootstrap-table/files/402903/request_before.txt)

**Comment by djhvscf** [comment]
Great! @Flounn

**Comment by djhvscf** [comment]
@wenzhixin can you take a look here?

**Comment by wenzhixin** [comment]
OK, I will test this

**Comment by Flounn** [comment]
I provide a jsFiddle => [http://jsfiddle.net/qye101xf/2/](http://jsfiddle.net/qye101xf/2/)

**Pull request closed without merging**